### PR TITLE
Update clang-format configuration to check java source files

### LINKED ISSUE #398: Apply format and code conventions to java source files
## Enhacement description

To ensure the best practices in `Java` code it would be convenient to use a tool that allows us to format the source code and check if a file (or files) is following the style/format rules. In addition, it would be necessary to create a new workflow to check the code in each push or pull request.

### PR DESCRIPTION
**Pull request by ManuelJNunez**
:white_check_mark: I have updated the documentation accordingly.
:white_check_mark: I have read the CONTRIBUTING document.

### Summary

I updated the clang-format configuration so that it checks java source files too.

### Details and comments

close #398

### CODE REVIEW

**Comment by lulivi** [comment]
I'm thinking that we may want to separate cxx and java files from the formatting steps, right? What do you think @ManuelJNunez and @angelrti

**Comment by ManuelJNunez** [comment]
@lulivi maybe I can use the separators in the same file. Take a look [here](https://clang.llvm.org/docs/ClangFormatStyleOptions.html).

File: .clang-format [hunk 1]
```diff
@@ -138,3 +138,113 @@ BreakBeforeBinaryOperators: NonAssignment
...
+# CFR3-05: Statements
+AllowShortBlocksOnASingleLine: false
+AllowShortFunctionsOnASingleLine: false
+
+# CFR3-06: Pointer declaration
+# Checkout CFR1-08
+
+BreakBeforeBraces: Custom
+BraceWrapping:
+    AfterClass: false             # CFR2-03: C classes
```
Comment (lulivi) [inline .clang-format:210]:
```suggestion
    AfterClass: false            # CFR2-03: C classes
```

**Comment by lulivi** [comment]
I'm not sure if we really want some duplicated code in the .clang-format file, because they are the same, right?

**Review by lulivi — APPROVED** [review]

File: resources/ci_cd/linux_format.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def get_argument_parser() -> ArgumentParser:
         help="Disable markdownlint linter.",
     )
     parser.add_argument(
-        "-cxx",
+        "-clang",
```
Comment (angelrti) [inline resources/ci_cd/linux_format.py:100]: we need to update the contributing.md accordingly

**Review by angelrti — CHANGES_REQUESTED** [review]

File: resources/ci_cd/linux_format.py (hunk 2 above)
Comment (ManuelJNunez) [reply resources/ci_cd/linux_format.py:100]: Done 👍🏻

**Review by angelrti — APPROVED** [review]

**Review by lulivi — APPROVED** [review]

**Pull request merged**